src/index.js

```diff
@@ -2,14 +2,29 @@
 
 exports.__esModule = true;
 exports.defineConfig = void 0;
+exports.defineFlatConfig = void 0;
 
 /**
- * Define an eslint config.
+ * Define an ESLint config.
  *
- * @param config Eslint config.
- * @returns Eslint config.
+ * @param config ESLint config.
+ * @returns ESLint config.
  */
 function defineConfig(config) {
   return config;
 }
+
+/**
+ * Define a flat ESLint config.
+ *
+ * @param config flat ESLint config.
+ * @returns flat ESLint config.
+ *
+ * @see [Configuration Files (New)](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new)
+ */
+function defineFlatConfig(config) {
+  return config;
+}
+
 exports.defineConfig = defineConfig;
+exports.defineFlatConfig = defineFlatConfig;
```

Comment:
Isn't it possible to use just `defineConfig` and manage the rest fully via TypeScript?
I think we can use function overloads, so when pass a `ESLintFlatConfig[]` the return value will be `ESLintFlatConfig[]` and when you pass `ESLintConfig` the return value will be `ESLintConfig`